### PR TITLE
Feature: Support for Custom Records

### PR DESCRIPTION
**Pull request by ashwin1111**

### CODE REVIEW

File: src/app/core/services/mappings.service.ts [hunk 1]
```diff
@@ -99,6 +100,25 @@ export class MappingsService {
     return this.fyleExpenseCustomFields;
   }
 
+  postNetsuiteExpenseCustomFields(sync: boolean = false) {
```
Comment (Shwetabhk) [inline src/app/core/services/mappings.service.ts:103]: Take care of naming changes

File: src/app/netsuite/settings/netsuite-configurations/custom-records/custom-records.component.html [hunk 2]
```diff
@@ -0,0 +1,61 @@
+<div class="internal-loader" fxLayoutAlign="center start" *ngIf="isLoading" fxFlexFill>
```
Comment (Shwetabhk) [inline src/app/netsuite/settings/netsuite-configurations/custom-records/custom-records.component.html:1]: check for zero state

File: src/app/netsuite/settings/netsuite-configurations/custom-records/custom-records.component.html [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+          Add Custom List / Record items from Netsuite.
+        </div>
+        <a href="https://www.fylehq.com/help/en/articles/4424248-onboarding-process-to-set-up-fyle-netsuite-integration"
+          target="_blank" rel="noopener noreferrer" class="read-more-link" fxLayout="row"
+          fxLayoutAlign="start center">Read
+          more<mat-icon class="read-more-link--icon">open_in_new</mat-icon></a>
+      </div>
+
+      <button class="page-info--primary-action" mat-flat-button color="primary" type="button" (click)="open()">
+        Create Custom List
```
Comment (Shwetabhk) [inline src/app/netsuite/settings/netsuite-configurations/custom-records/custom-records.component.html:17]: `Add Custom List / Record`

File: src/app/netsuite/settings/netsuite-configurations/custom-records/custom-records.component.html [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+      <zero-state message="You have no employee mappings yet"></zero-state>
+    </div>
+  
+    <div *ngIf="customMappings.length" [hidden]="isLoading">
+      <table mat-table [dataSource]="customMappings" class="page-table" matSort matSortActive="created_at"
+        matSortDisableClear matSortDirection="desc">
+  
+        <ng-container matColumnDef="script_id">
+          <th mat-header-cell *matHeaderCellDef>
+            Transaction Custom Field ScriptId
```
Comment (Shwetabhk) [inline src/app/netsuite/settings/netsuite-configurations/custom-records/custom-records.component.html:32]: NetSuite Script Id

File: src/app/netsuite/settings/netsuite-configurations/custom-records/custom-records.component.html [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+          </th>
+          <td mat-cell *matCellDef="let element">
+            {{element.script_id}}
+          </td>
+        </ng-container>
+  
+  
+        <ng-container matColumnDef="internal_id">
+          <th mat-header-cell *matHeaderCellDef>
+            Custom Record / Custom Field Internal Id
```
Comment (Shwetabhk) [inline src/app/netsuite/settings/netsuite-configurations/custom-records/custom-records.component.html:42]: Custom Record / Field Internal Id

File: src/app/netsuite/settings/netsuite-configurations/custom-records/custom-records.component.html [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+          </th>
+          <td mat-cell *matCellDef="let element">
+            {{ element.internal_id }}
+          </td>
+        </ng-container>
+
+        <ng-container matColumnDef="type">
+          <th mat-header-cell *matHeaderCellDef> Custom Field type</th>
```
Comment (Shwetabhk) [inline src/app/netsuite/settings/netsuite-configurations/custom-records/custom-records.component.html:50]: Custom Field Type

File: src/app/netsuite/settings/netsuite-configurations/custom-records/custom-records-dialog/custom-records-dialog.component.ts [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+  submit() {
+    const that = this;
+    that.isLoading = true;
+
+    const customFields = {
+      internal_id: that.form.value.internal_id,
+      script_id: that.form.value.script_id,
+      custom_type: that.form.value.custom_field_type
+    };
+
```
Comment (Shwetabhk) [inline src/app/netsuite/settings/netsuite-configurations/custom-records/custom-records-dialog/custom-records-dialog.component.ts:54]: change records to segments in folder names

File: src/app/netsuite/settings/netsuite-configurations/netsuite-configurations.component.html [hunk 8]
```diff
@@ -15,6 +15,9 @@
       <a mat-tab-link (click)="changeState('EXPENSE_FIELDS')" *ngIf="showExpenseFields()" [active]="state==='EXPENSE_FIELDS'">
         Expense Fields
       </a>
+      <a mat-tab-link (click)="changeState('CUSTOM_RECORDS')" *ngIf="showExpenseFields()" [active]="state==='CUSTOM_RECORDS'">
+        Netsuite Custom Records
```
Comment (Shwetabhk) [inline src/app/netsuite/settings/netsuite-configurations/netsuite-configurations.component.html:19]: NetSuite Expense Fields

**Comment by ashwin1111** [comment]
New Tab
Screenshot from 2020-10-27 17-47-19

**Comment by ashwin1111** [comment]
Screenshot from 2020-10-27 17-48-01

**Comment by ashwin1111** [comment]
Screenshot from 2020-10-27 17-48-56

**Review by Shwetabhk — APPROVED** [review]

**Pull request merged**